### ISSUE TITLE
Borked H(div) trace tabulation when asking for derivatives

### ISSUE THREAD

**Issue by miklos1** [firedrakeproject/fiat#7]
```python
import FIAT
from FIAT.reference_element import UFCTetrahedron

elem = FIAT.HDivTrace(UFCTetrahedron(), 0)
print elem.tabulate(1, [[1./3, 1./3]], entity=(2, 0))
```
prints:
```
{(1, 0, 0): TraceError('Gradient evaluations are illegal on trace elements.',),
 (0, 1, 0): TraceError('Gradient evaluations are illegal on trace elements.',),
 (0, 0, 0): TraceError('Gradient evaluations are illegal on trace elements.',),
 (0, 0, 1): array([[ 1.],
       [ 0.],
       [ 0.],
       [ 0.]])}
```

**Comment by thomasgibson** [comment]
Thanks for the issue. I'll have a close look at this tomorrow.

**Comment by thomasgibson** [comment]
This is being addressed on Bitbucket: [FIAT PR #31](https://bitbucket.org/fenics-project/fiat/pull-requests/31/gradient-tabulation-bug-fix-for-the-trace/diff).
